examples/evm_bots.py

```diff
@@ -443,220 +490,300 @@ def create_agent(
     """
-    dev_accounts: list[KeyfileAccount] = get_accounts(config)
+    dev_accounts: list[KeyfileAccount] = get_accounts(experiment_config)
     faucet = None
-    if not config.scratch["devnet"]:
-        faucet = ape_utils.get_instance(FAUCET_ADDRESS, provider=provider)
+    if not experiment_config.scratch["devnet"]:
+        faucet = ape_utils.get_instance(
+            experiment_config.scratch["goerli_faucet_address"], provider=experiment_config.scratch["provider"]
+        )
     bot_num = 0
-    for bot_name in config.scratch["bot_names"]:
-        _policy = config.scratch[bot_name].policy
+    for bot_name in experiment_config.scratch["bot_names"]:
+        _policy = experiment_config.scratch[bot_name].policy
         log_and_show(
-            f"{bot_name:6s}: n={config.scratch[f'num_{bot_name}']}  "
+            f"{bot_name:6s}: n={experiment_config.scratch[f'num_{bot_name}']}  "
             f"policy={_policy.__name__ if _policy.__module__ == '__main__' else _policy.__module__:20s}"
         )
-        bot_num += config.scratch[f"num_{bot_name}"]
+        bot_num += experiment_config.scratch[f"num_{bot_name}"]
     _sim_agents = {}
     start_time_ = now()
     on_chain_trade_info: ape_utils.OnChainTradeInfo = ape_utils.get_on_chain_trade_info(
-        hyperdrive_contract=hyperdrive_contract
+        hyperdrive_contract=experiment_config.scratch["hyperdrive_instance"]
     )
     log_and_show(f"Getting on-chain trade info took {fmt(now() - start_time_)} seconds")
-    for bot_name in [name for name in config.scratch["bot_names"] if config.scratch[f"num_{name}"] > 0]:
-        bot_info = config.scratch[bot_name]
+    for bot_name in [
+        name for name in experiment_config.scratch["bot_names"] if experiment_config.scratch[f"num_{name}"] > 0
+    ]:
+        bot_info = experiment_config.scratch[bot_name]
         bot_info.name = bot_name
-        for _ in range(config.scratch[f"num_{bot_name}"]):  # loop across number of bots of this type
+        for _ in range(experiment_config.scratch[f"num_{bot_name}"]):  # loop across number of bots of this type
             bot_info.index = len(_sim_agents)
             logging.debug("Creating %s agent %s/%s: %s", bot_name, bot_info.index + 1, bot_num, bot_info)
             agent = create_agent(
                 bot=bot_info,
                 dev_accounts=dev_accounts,
                 faucet=faucet,
-                base_=base_contract,
+                base_=experiment_config.scratch["base_instance"],
                 on_chain_trade_info=on_chain_trade_info,
-                hyperdrive_contract=hyperdrive_contract,
-                config=experiment_config,
+                hyperdrive_contract=experiment_config.scratch["hyperdrive_instance"],
+                experiment_config=experiment_config,
             )
             _sim_agents[f"agent_{agent.wallet.address}"] = agent
-    return _sim_agents
+    experiment_config.scratch["sim_agents"] = _sim_agents
```

Comment:
We should restrict `config` variables to be just that, configurations for the experiment. It should be easily summarized or logged. By putting more complicated objects, such as the simulation agents, into the scratch, you end up with a pattern that we do not want in practice. Specifically, the `config.scratch` dict shouldn't end up a catch-all holding anything we want to pass from function-to-function.